inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,308 @@
+        if not record.get(key): # adding to object as well
+            creation_keys = self.keys[position:]
+            record.update(create_schema_record(schema, creation_keys, self.value))
+            return
+        if position + 1 == len(self.keys):
+            if isinstance(record[key], list):
+                record[key].append(self.value)
+        else:
+            if isinstance(record[key], list):
+                for array_record in record[key]:
+                    self.apply_action(array_record, new_schema, position+1, checked)
+            else:
+                self.apply_action(record[key], new_schema, position+1, checked)
+
+
+class Deletion(object):
+    def __init__(self, keys, value, update_regex, values_to_check,
+                 where_regex, where_keys, where_values):
+        self.keys = keys
+        self.value = value
+        self.update_regex = update_regex
+        self.values_to_check = values_to_check
+        self.where_regex = where_regex
+        self.where_keys = where_keys
+        self.where_values = where_values
+
+    def apply_action(self, record, schema, position=0, checked=False):
+        """Recursive function to change a record object."""
+        key = self.keys[position]
+        where_key = None
+        if len(self.where_keys) > position:
+            where_key = self.where_keys[position]
+        new_schema = {}
+        if schema: # for testing purposes
+            if schema['type'] == 'object':
+                new_schema = schema['properties'][key]
+            elif schema['type'] == 'array':
+                new_schema = schema['items']['properties'][key]
+
+        if where_key:
+            if not checked and key != where_key: # fixme possible refactor
+                if not check_value(record=record, keys=self.where_keys, values_to_check=self.where_values,
+                                   regex=self.where_regex, position=position):
+                    return
+                checked = True
+        if not record.get(key):
+            return
+        if position + 1 == len(self.keys):
+            if isinstance(record[key], list):
+                if self.update_regex:
+                    for value_to_check in self.values_to_check:
+                        record[key] = list(
+                            filter(lambda x: (re.search(
+                                re.escape(value_to_check),
+                                x) or len(self.values_to_check) == 0),
+                                   record[key]))
+                else:
+                    record[key] = list(
+                        filter(lambda x: not (len(self.values_to_check) == 0
+                                              or x in self.values_to_check), record[key]))
+            else:
+                if len(self.values_to_check) == 0 \
+                        or record[key] in self.values_to_check:
+                    del record[key]  # Mark for deletion
+                if self.update_regex:
```

Comment:
Are `values_to_check` and `update_regex` incompatible? If you have both and you delete the key above then here you will get a key error in `record[key]`